tree_sitter/binding.c

```diff
@@ -15,6 +15,8 @@ typedef struct {
 typedef struct {
   PyObject_HEAD
   TSTree *tree;
+  PyObject *source;
+  int edited;
```

Comment:
Could we just release the `source` and set it to None when editing the tree, instead of having a separate `edited` flag? This seems  simpler, since we already have to check if the source is `Py_None` when reading the text of a node.

If we did it that way, I don't think we'd even need to use `tree_get_text` inside of `node_get_text`, and we wouldn't need to forward-declare that function.

Reply:
Thanks for the feedback! This makes a lot of sense indeed. I have done the proposed changes, let me know what you think!